public/app/plugins/datasource/elasticsearch/datasource.ts

```diff
@@ -919,9 +923,10 @@ export class ElasticDatasource
 
         const buckets = res.responses[0].aggregations['1'].buckets;
         return _map(buckets, (bucket) => {
+          const keyString = bucket.key_as_string || String(bucket.key);
           return {
-            text: bucket.key_as_string || bucket.key,
-            value: bucket.key,
+            text: keyString,
+            value: isTagValueQuery ? keyString : bucket.key,
```

Comment:
Technically `MetricFindValue` is supposed to accept numbers as values, but it was logging a console error instead of adding the filter when I tried it, it worked properly when it was a string.